<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Quality Control of Peptide Data with the <code>pmartR</code> Package • pmartR</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Quality Control of Peptide Data with the &lt;code&gt;pmartR&lt;/code&gt; Package">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">pmartR</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">0.9.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Quality_Control_with_pmartR.html">Quality Control of Peptide Data with the `pmartR` Package</a>
    </li>
    <li>
      <a href="../articles/Statistical_Analysis_with_pmartR.html">Statistical Analysis with the `pmartR` Package</a>
    </li>
    <li>
      <a href="../articles/analysis_log.html">analysis_log</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/pmartR/pmartR">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Quality Control of Peptide Data with the <code>pmartR</code> Package</h1>
                        <h4 class="author">Kelly Stratton, Lisa Bramer, Iobani Godinez</h4>
            
            <h4 class="date">2018-08-24</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/pmartR/pmartR/blob/master/vignettes/Quality_Control_with_pmartR.Rmd"><code>vignettes/Quality_Control_with_pmartR.Rmd</code></a></small>
      <div class="hidden name"><code>Quality_Control_with_pmartR.Rmd</code></div>

    </div>

    
    
<p><img src="pmartR_logo_final.jpg" width="100px"></p>
<pre><code>## Loading required package: data.table</code></pre>
<p>The pmartR package provides functionality for quality control (QC) processing of mass spectrometry (MS) pan-omics data, in particular proteomic (either at the peptide or the protein level), lipidomic, and metabolomic data. This includes data transformation, specification of groups that are to be compared against each other, filtering of features and/or samples, data normalization, and data summarization (correlation, principal components analysis).</p>
<p>Sample datasets are available in the pmartRdata package for each of the following data types: peptide level data, protein level data, lipidomic data, and metabolomic data. Under development are 3 additional packages that will contain functionality for processing MS omics data. These are pmartRstat for statistical analysis, pmartRpq for protein quantification of peptide level data, and pmartRviz for visualization of the data incorporating the statistical results.</p>
<p>This vignette focuses on the quality control of proteomic data at the peptide level. Other vignettes will be available shortly describing analysis of the other data types.</p>
<div id="data" class="section level2">
<h2 class="hasAnchor">
<a href="#data" class="anchor"></a>Data</h2>
<p>Below we load in 3 data.frames from the pmartRdata package:</p>
<ul>
<li><p><code>pep_edata</code>: A <span class="math inline">\(p * (n + 1)\)</span> data.frame of expression data, where <span class="math inline">\(p\)</span> is the number of peptides observed and <span class="math inline">\(n\)</span> is the number of samples (an additional peptide identifier/name column should also be present anywhere in the data.frame). Each row corresponds to data for each peptide.</p></li>
<li><p><code>pep_fdata</code>: A data.frame with <span class="math inline">\(n\)</span> rows. Each row corresponds to a sample with one column giving the unique sample identifiers found in e_data column names and other columns providing qualitative and/or quantitative traits of each sample.</p></li>
<li><p><code>pep_emeta</code>: An optional data.frame with at least <span class="math inline">\(p\)</span> rows. Each row corresponds to a peptide with one column giving peptide names (must be named the same as the column in <code>e_data</code>) and other columns giving meta information (e.g. mappings of peptides to proteins).</p></li>
</ul>
<p>Here, <span class="math inline">\(p\)</span> is 17407 and <span class="math inline">\(n\)</span> is 12. The <code>pep_fdata</code> data.frame contains the sample identifier and the condition designation for each sample, either ‘Infection’ (samples that were infected with some pathogen) or ‘Mock’ (uninfected samples used as a control). The <code>pep_emeta</code> data.frame contains the peptide identifier column (‘Mass_Tag_ID’), the protein identification name to which each mass tag ID maps (‘Protein’), a reference identifier (‘Ref_ID’), and the peptide sequence (Peptide_Sequence). Note that the mass tag IDs are unique (<span class="math inline">\(n_{peptide} =\)</span> 17407) while the protein identifiers are not (<span class="math inline">\(n_{protein} =\)</span> 2774) since multiple peptides often map to the same protein.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">data</span>(<span class="st">"pep_edata"</span>)
<span class="kw">dim</span>(pep_edata)</code></pre></div>
<pre><code>## [1] 17407    13</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">pep_edata[<span class="dv">1</span><span class="op">:</span><span class="dv">6</span>,<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>]</code></pre></div>
<pre><code>##    Mass_Tag_ID Infection1 Infection2 Infection3 Infection4
## 4         1024         NA         NA         NA         NA
## 6         1047   17953839   20071472   20745779   18206556
## 9         1055  109536335  115459820  106127139   74522014
## 14        1104 1752288782 1796561709 1703186182 2438218572
## 16        1110    2571804    4269824    4852871    2630414
## 23        1214  110239193   82436688  100447189  102006001</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">data</span>(<span class="st">"pep_fdata"</span>)
<span class="kw">dim</span>(pep_fdata)</code></pre></div>
<pre><code>## [1] 12  2</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(pep_fdata)</code></pre></div>
<pre><code>##     SampleID Condition
## 1 Infection1 Infection
## 2 Infection2 Infection
## 3 Infection3 Infection
## 4 Infection4 Infection
## 5 Infection5 Infection
## 6 Infection6 Infection</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">data</span>(<span class="st">"pep_emeta"</span>)
<span class="kw">dim</span>(pep_emeta)</code></pre></div>
<pre><code>## [1] 17407     4</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(pep_emeta)</code></pre></div>
<pre><code>##    Mass_Tag_ID    Protein Ref_ID         Peptide_Sequence
## 6         1024 ACBP_HUMAN    324 K.QATVGDINTERPGMLDFTGK.A
## 15        1047 ALBU_HUMAN    580       T.ECCHGDLLECADDR.A
## 20        1055 RL40_HUMAN   9898     K.TITLEVEPSDTIENVK.A
## 35        1104 ALBU_HUMAN    580      K.KVPQVSTPTLVEVSR.N
## 39        1110  CYC_HUMAN   2918          K.TGPNLHGLFGR.K
## 61        1214  G3P_HUMAN   4480             K.VGVNGFGR.I</code></pre>
<p>Next we create pepData object, which contains the 3 components loaded above. During object creation, we specify the names of the identifier columns in the <code>e_data</code>, <code>f_data</code>, and <code>e_meta</code> components of the data (<code>edata_cname = "Mass_Tag_ID", emeta_cname = "Protein", fdata_cname = "SampleID"</code>, respectively), as well as the scale of the data (<code>data_scale = "abundance"</code>). We can use the <code>summary</code> function to get basic summary information for our pepData object.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">mypepData &lt;-<span class="st"> </span><span class="kw"><a href="../reference/as.pepData.html">as.pepData</a></span>(<span class="dt">e_data =</span> pep_edata, <span class="dt">f_data =</span> pep_fdata, <span class="dt">e_meta =</span> pep_emeta, <span class="dt">edata_cname =</span> <span class="st">"Mass_Tag_ID"</span>, <span class="dt">emeta_cname =</span> <span class="st">"Protein"</span>, <span class="dt">fdata_cname =</span> <span class="st">"SampleID"</span>, <span class="dt">data_scale =</span> <span class="st">"abundance"</span>)
<span class="kw">class</span>(mypepData)</code></pre></div>
<pre><code>## [1] "pepData"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">summary</span>(mypepData)</code></pre></div>
<pre><code>##                                     
## Class                        pepData
## Unique SampleIDs (f_data)         12
## Unique Mass_Tag_IDs (e_data)   17407
## Unique Proteins (e_meta)        2774
## Missing Observations           82833
## Proportion Missing             0.366</code></pre>
<p>Alternatively, the pmartRdata package contains the corresponding pepData object, which can be loaded as follows.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">data</span>(<span class="st">"pep_object"</span>)
<span class="kw">class</span>(pep_object)</code></pre></div>
<pre><code>## [1] "pepData"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">rm</span>(pep_object)</code></pre></div>
<p>We can use the <code>plot()</code> function to display box plots for each sample in the pepData object.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(mypepData)</code></pre></div>
<p><img src="Quality_Control_with_pmartR_files/figure-html/data4-1.png" width="480"></p>
<div id="additional-data" class="section level3">
<h3 class="hasAnchor">
<a href="#additional-data" class="anchor"></a>Additional Data</h3>
<p>Also included in the pmartR package are protein, metabolite, and lipid data, in analagous formats to the peptide data described above.</p>
<p>Finally, we include .csv versions of the metabolite data (e_data and f_data), where the sample names have been modified to begin with numbers and contain dashes. These are included to illustrate use of the “check.names” argument to the  (and ) functions. If we read in the e_data file without specifying , R automatically puts an “X” at the beginning of the sample names so they do not begin with a number and replaces the dashes with periods. This causes the sample names in e_data to not match the sample names in f_data, and will prevent the creation of a metabData object.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">edata &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="kw">system.file</span>(<span class="st">"extdata"</span>, <span class="st">"metab_edata_sample_names.csv"</span>, <span class="dt">package=</span><span class="st">"pmartRdata"</span>), <span class="dt">header=</span><span class="ot">TRUE</span>)
<span class="kw">names</span>(edata)</code></pre></div>
<pre><code>##  [1] "Metabolite"   "X1.Mock"      "X2.Mock"      "X3.Mock"     
##  [5] "X1.Infection" "X2.Infection" "X3.Infection" "X4.Infection"
##  [9] "X5.Infection" "X6.Infection" "X7.Infection" "X8.Infection"
## [13] "X9.Infection"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">fdata &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="kw">system.file</span>(<span class="st">"extdata"</span>, <span class="st">"metab_fdata_sample_names.csv"</span>, <span class="dt">package=</span><span class="st">"pmartRdata"</span>), <span class="dt">header=</span><span class="ot">TRUE</span>)

<span class="co"># Do the sample names match each other? #</span>
<span class="kw">all</span>(<span class="kw">names</span>(edata)[<span class="op">-</span><span class="dv">1</span>] <span class="op">==</span><span class="st"> </span>fdata<span class="op">$</span>SampleID)</code></pre></div>
<pre><code>## [1] FALSE</code></pre>
<p>To avoid this, without having to modify the sample names in f_data, we can specify :</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">edata &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="kw">system.file</span>(<span class="st">"extdata"</span>, <span class="st">"metab_edata_sample_names.csv"</span>, <span class="dt">package=</span><span class="st">"pmartRdata"</span>), <span class="dt">header=</span><span class="ot">TRUE</span>, <span class="dt">check.names=</span><span class="ot">FALSE</span>)
<span class="kw">names</span>(edata)</code></pre></div>
<pre><code>##  [1] "Metabolite"  "1-Mock"      "2-Mock"      "3-Mock"      "1-Infection"
##  [6] "2-Infection" "3-Infection" "4-Infection" "5-Infection" "6-Infection"
## [11] "7-Infection" "8-Infection" "9-Infection"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Do the sample names match each other? #</span>
<span class="kw">all</span>(<span class="kw">names</span>(edata)[<span class="op">-</span><span class="dv">1</span>] <span class="op">==</span><span class="st"> </span>fdata<span class="op">$</span>SampleID)</code></pre></div>
<pre><code>## [1] TRUE</code></pre>
</div>
</div>
<div id="workflow" class="section level2">
<h2 class="hasAnchor">
<a href="#workflow" class="anchor"></a>Workflow</h2>
<p>Once the pepData object is created, a typical QC workflow follows the figure below.</p>
<div class="figure">
<img src="pmartR_graphic-final.jpg" alt="Figure 1. Quality control and processing workflow in pmartR package." width="600px"><p class="caption">
Figure 1. Quality control and processing workflow in pmartR package.
</p>
</div>
<p>We will walk through the steps in this QC workflow using <code>mypepData</code>.</p>
<div id="format-data" class="section level3">
<h3 class="hasAnchor">
<a href="#format-data" class="anchor"></a>Format Data</h3>
<p>Sometimes omics abundance data contains 0s when the particular biomolecule was not detected for a given sample. It is our practice to replace any 0s with NAs as opposed to imputing the missing values. We strongly prefer NAs to imputation because we cannot know whether the biomolecule is not present in the sample, simply was below the limit of detection, or was present but missing from our data at random. The function <code>edata_replace()</code> can be used to replace values in the pepData object.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">mypepData &lt;-<span class="st"> </span><span class="kw">edata_replace</span>(mypepData, <span class="dt">x =</span> <span class="dv">0</span>, <span class="dt">y =</span> <span class="ot">NA</span>)</code></pre></div>
<pre><code>## 0 instances of 0 have been replaced with NA</code></pre>
<p>In this dataset, there were no 0s so nothing was replaced.</p>
<p>We also highly recommend log transforming the data prior to analysis. pmartR supports log2, log10, and natural logarithm transformations. The <code>edata_transform()</code> function provides this capability. We can also use <code>edata_trasnform()</code> to transform the data back to the abundance scale if needed. Note that the scale of the data is stored and automatically updated in the <code>data_info$data_scale</code> attribute of the pepData object. Below we log10 transform the data, then back transform to the abundance scale, and finally settle on the log2 scale.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">mypepData &lt;-<span class="st"> </span><span class="kw">edata_transform</span>(mypepData, <span class="dt">data_scale =</span> <span class="st">"log10"</span>)
<span class="kw">attributes</span>(mypepData)<span class="op">$</span>data_info<span class="op">$</span>data_scale</code></pre></div>
<pre><code>## [1] "log10"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">mypepData &lt;-<span class="st"> </span><span class="kw">edata_transform</span>(mypepData, <span class="dt">data_scale =</span> <span class="st">"abundance"</span>)
<span class="kw">attributes</span>(mypepData)<span class="op">$</span>data_info<span class="op">$</span>data_scale</code></pre></div>
<pre><code>## [1] "abundance"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">mypepData &lt;-<span class="st"> </span><span class="kw">edata_transform</span>(mypepData, <span class="dt">data_scale =</span> <span class="st">"log2"</span>)
<span class="kw">attributes</span>(mypepData)<span class="op">$</span>data_info<span class="op">$</span>data_scale</code></pre></div>
<pre><code>## [1] "log2"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(mypepData)</code></pre></div>
<p><img src="Quality_Control_with_pmartR_files/figure-html/format2-1.png" width="480"></p>
<p>Finally, we are preparing this data for statistical analysis where we will compare the samples belonging to one group to the samples belonging to another, and so we must specify the group membership of the samples. We do this using the <code><a href="../reference/group_designation.html">group_designation()</a></code> function, which modifies our pepData object and returns an updated version of it. Up to two main effects and up to two covariates may be specified, with one main effect being required at minimum. For the example data, we only have one option - specify the main effect to be “Condition” - since we do not have any additional information about the samples. Certain functions we will use below require that groups have been designated via the <code><a href="../reference/group_designation.html">group_designation()</a></code> function.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">mypepData &lt;-<span class="st"> </span><span class="kw"><a href="../reference/group_designation.html">group_designation</a></span>(mypepData, <span class="dt">main_effects =</span> <span class="st">"Condition"</span>, <span class="dt">covariates =</span> <span class="ot">NULL</span>)</code></pre></div>
<p>The <code><a href="../reference/group_designation.html">group_designation()</a></code> function creates an attribute of the dataset as follows:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">attributes</span>(mypepData)<span class="op">$</span>group_DF</code></pre></div>
<pre><code>##      SampleID     Group
## 1  Infection1 Infection
## 2  Infection2 Infection
## 3  Infection3 Infection
## 4  Infection4 Infection
## 5  Infection5 Infection
## 6  Infection6 Infection
## 7  Infection7 Infection
## 8  Infection8 Infection
## 9  Infection9 Infection
## 10      Mock1      Mock
## 11      Mock2      Mock
## 12      Mock3      Mock</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(mypepData)</code></pre></div>
<p><img src="Quality_Control_with_pmartR_files/figure-html/format4-1.png" width="480"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(mypepData, <span class="dt">color_by =</span> <span class="st">"Condition"</span>, <span class="dt">bw_theme=</span><span class="ot">TRUE</span>)</code></pre></div>
<p><img src="Quality_Control_with_pmartR_files/figure-html/format4-2.png" width="480"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(mypepData, <span class="dt">color_by =</span> <span class="st">"Condition"</span>, <span class="dt">title_size=</span><span class="dv">10</span>, <span class="dt">legend_position=</span><span class="st">"bottom"</span>)</code></pre></div>
<p><img src="Quality_Control_with_pmartR_files/figure-html/format4-3.png" width="480"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(mypepData, <span class="dt">facet_by =</span> <span class="st">"Condition"</span>, <span class="dt">color_by =</span> <span class="st">"Condition"</span>)</code></pre></div>
<p><img src="Quality_Control_with_pmartR_files/figure-html/format4-4.png" width="480"></p>
</div>
<div id="filter-peptides" class="section level3">
<h3 class="hasAnchor">
<a href="#filter-peptides" class="anchor"></a>Filter Peptides</h3>
<p>It is often good practice to filter out peptides that do not meet certain criteria, and we offer 5 different filters. Each of the filtering functions calculates metric(s) that can be used to filter out peptides and returns an S3 object. Using the <code>summary()</code> function produces a summary of the metric(s) and using the <code>plot()</code> function produces a graph. Filters that require a user-specified threshold in order to actually filter out peptides have corresponding <code>summary</code> and <code>plot</code> methods that take optional argument(s) to specify that threshold. Once one of the 5 peptide level filter functions has been called, the results of that function can be used in conjunction with the <code>applyFilt()</code> function to actually filter out peptides based on the metric(s) and user-specified threshold(s) and create a new, filtered pepData object.</p>
<div id="proteomics-filter" class="section level4">
<h4 class="hasAnchor">
<a href="#proteomics-filter" class="anchor"></a>Proteomics Filter</h4>
<p>The proteomics filter is applicable only to peptide level data that contains the <code>e_meta</code> component, as it counts the number of peptides that map to each protein and/or the number of proteins to which each individual peptide maps. It returns a list of two character vectors, the first, <code>peptides_filt</code>, giving degenerate peptide names. The second, <code>proteins_filt</code>, gives the names of proteins which no longer have peptides mapping to them in the dataset. This filter does not require a user-specified threshold.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">myfilter &lt;-<span class="st"> </span><span class="kw"><a href="../reference/proteomics_filter.html">proteomics_filter</a></span>(mypepData)
<span class="kw">summary</span>(myfilter)</code></pre></div>
<pre><code>## 
## Summary of Proteomics Filter 
## 
##              Obs. Per Peptide Obs. Per Protein
## Min.                        1                1
## 1st Qu.                     1                1
## Median                      1                3
## Mean                        1 6.27505407354001
## 3rd Qu.                     1                7
## Max.                        1              333
##                                               
## Filtered                    0                0
## Not Filtered            17407             2774</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(myfilter)</code></pre></div>
<p><img src="Quality_Control_with_pmartR_files/figure-html/proteomics_filter-1.png" width="480"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(myfilter, <span class="dt">bw_theme=</span><span class="ot">TRUE</span>)</code></pre></div>
<p><img src="Quality_Control_with_pmartR_files/figure-html/proteomics_filter-2.png" width="480"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(myfilter, <span class="dt">title_size=</span><span class="dv">10</span>, <span class="dt">x_lab_size=</span><span class="dv">9</span>, <span class="dt">y_lab_size=</span><span class="dv">9</span>)</code></pre></div>
<p><img src="Quality_Control_with_pmartR_files/figure-html/proteomics_filter-3.png" width="480"></p>
<p>With the current dataset, there is a one-to-one mapping between peptides and proteins, so these plots are not very informative. With a many-to-one mapping, we’d see a histogram.</p>
</div>
<div id="molecule-filter" class="section level4">
<h4 class="hasAnchor">
<a href="#molecule-filter" class="anchor"></a>Molecule Filter</h4>
<p>The molecule filter allows the user to remove from the dataset any biomolecule not seen in at least <code>min_num</code> of the samples. The user may specify a threshold of the minimum number of times each biomolecule must be observed across all samples; the default value is 2.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">myfilter &lt;-<span class="st"> </span><span class="kw"><a href="../reference/molecule_filter.html">molecule_filter</a></span>(mypepData)
<span class="kw">plot</span>(myfilter)</code></pre></div>
<p><img src="Quality_Control_with_pmartR_files/figure-html/molecule_filter-1.png" width="480"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">summary</span>(myfilter, <span class="dt">min_num =</span> <span class="dv">3</span>)</code></pre></div>
<pre><code>## 
## Summary of Molecule Filter 
## ----------------------------------                                  
## Minimum Number:                  3
## Filtered:                     3342
## Not Filtered:                14065
##                                   
## Molecule Observation Counts:      
## 
##  num_observations frequency_counts
##                 1             1814
##                 2             3342
##                 3             4911
##                 4             5934
##                 5             6737
##                 6             7530
##                 7             8389
##                 8             9279
##                 9            10383
##                10            11527
##                11            12987
##                12            17407</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(myfilter, <span class="dt">min_num =</span> <span class="dv">3</span>)</code></pre></div>
<p><img src="Quality_Control_with_pmartR_files/figure-html/molecule_filter-2.png" width="480"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(myfilter, <span class="dt">min_num =</span> <span class="dv">3</span>, <span class="dt">bw_theme =</span> <span class="ot">TRUE</span>)</code></pre></div>
<p><img src="Quality_Control_with_pmartR_files/figure-html/molecule_filter-3.png" width="480"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(myfilter, <span class="dt">min_num =</span> <span class="dv">3</span>, <span class="dt">title_size =</span> <span class="dv">10</span>, <span class="dt">x_lab_size =</span> <span class="dv">6</span>, <span class="dt">y_lab_size =</span> <span class="dv">8</span>)</code></pre></div>
<p><img src="Quality_Control_with_pmartR_files/figure-html/molecule_filter-4.png" width="480"></p>
<p>Maybe we don’t want to remove as many peptides as the above would, so we can look at <code>min_num=2</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">summary</span>(myfilter, <span class="dt">min_num =</span> <span class="dv">2</span>)</code></pre></div>
<pre><code>## 
## Summary of Molecule Filter 
## ----------------------------------                                  
## Minimum Number:                  2
## Filtered:                     1814
## Not Filtered:                15593
##                                   
## Molecule Observation Counts:      
## 
##  num_observations frequency_counts
##                 1             1814
##                 2             3342
##                 3             4911
##                 4             5934
##                 5             6737
##                 6             7530
##                 7             8389
##                 8             9279
##                 9            10383
##                10            11527
##                11            12987
##                12            17407</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(myfilter, <span class="dt">min_num =</span> <span class="dv">2</span>)</code></pre></div>
<p><img src="Quality_Control_with_pmartR_files/figure-html/molecule_filter2-1.png" width="480"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">mypepData &lt;-<span class="st"> </span><span class="kw">applyFilt</span>(<span class="dt">filter_object =</span> myfilter, mypepData, <span class="dt">min_num =</span> <span class="dv">2</span>)
<span class="kw">summary</span>(mypepData)</code></pre></div>
<pre><code>##                                     
## Class                        pepData
## Unique SampleIDs (f_data)         12
## Unique Mass_Tag_IDs (e_data)   15593
## Unique Proteins (e_meta)        2511
## Missing Observations           62879
## Proportion Missing              0.31
## Samples per group: Infection       9
## Samples per group: Mock            3</code></pre>
</div>
<div id="coefficient-of-variation-filter" class="section level4">
<h4 class="hasAnchor">
<a href="#coefficient-of-variation-filter" class="anchor"></a>Coefficient of Variation Filter</h4>
<p>The coefficient of variation (CV) filter calculates the pooled CV values <span class="citation">(Ahmed 1995)</span> as</p>
<p><span class="math inline">\(CV=()\)</span></p>
<p>The user can then specify a CV threshold, above which peptides are removed.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">myfilter &lt;-<span class="st"> </span><span class="kw"><a href="../reference/cv_filter.html">cv_filter</a></span>(mypepData)
<span class="kw">plot</span>(myfilter)</code></pre></div>
<pre><code>## Warning in .plot.cvFilt(filter_object, cv_threshold, ...): For display
## purposes, biomolecules with pooled CV greater than 91.95 are not displayed
## in the graph. This corresponds to 153 biomolecules.</code></pre>
<pre><code>## Warning: Removed 153 rows containing non-finite values (stat_bin).</code></pre>
<p><img src="Quality_Control_with_pmartR_files/figure-html/cv_filter-1.png" width="480"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(myfilter, <span class="dt">bw_theme =</span> <span class="ot">TRUE</span>)</code></pre></div>
<pre><code>## Warning in .plot.cvFilt(filter_object, cv_threshold, ...): For display
## purposes, biomolecules with pooled CV greater than 91.95 are not displayed
## in the graph. This corresponds to 153 biomolecules.

## Warning in .plot.cvFilt(filter_object, cv_threshold, ...): Removed 153 rows
## containing non-finite values (stat_bin).</code></pre>
<p><img src="Quality_Control_with_pmartR_files/figure-html/cv_filter-2.png" width="480"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">summary</span>(myfilter, <span class="dt">cv_threshold =</span> <span class="dv">150</span>)</code></pre></div>
<pre><code>## 
## Summary of Coefficient of Variation (CV) Filter
## ----------------------
## CVs:
## 
##      Min.   1st Qu.    Median      Mean   3rd Qu.      Max. 
##   0.04778  24.85999  32.28337  34.83979  42.00244 168.69275 
## 
## Total NAs: 359
## Total Non-NAs: 15234 
## 
## Number Filtered Biomolecules: 2</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(myfilter, <span class="dt">cv_threshold =</span> <span class="dv">150</span>, <span class="dt">title_size =</span> <span class="dv">15</span>)</code></pre></div>
<pre><code>## Warning in .plot.cvFilt(filter_object, cv_threshold, ...): For display
## purposes, biomolecules with pooled CV greater than 91.95 are not displayed
## in the graph. This corresponds to 153 biomolecules.

## Warning in .plot.cvFilt(filter_object, cv_threshold, ...): Removed 153 rows
## containing non-finite values (stat_bin).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_rect).</code></pre>
<p><img src="Quality_Control_with_pmartR_files/figure-html/cv_filter-3.png" width="480"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">mypepData &lt;-<span class="st"> </span><span class="kw">applyFilt</span>(<span class="dt">filter_object =</span> myfilter, mypepData, <span class="dt">cv_threshold =</span> <span class="dv">150</span>)
<span class="kw">summary</span>(mypepData)</code></pre></div>
<pre><code>##                                     
## Class                        pepData
## Unique SampleIDs (f_data)         12
## Unique Mass_Tag_IDs (e_data)   15591
## Unique Proteins (e_meta)        2511
## Missing Observations           62871
## Proportion Missing              0.31
## Samples per group: Infection       9
## Samples per group: Mock            3</code></pre>
</div>
<div id="custom-filter" class="section level4">
<h4 class="hasAnchor">
<a href="#custom-filter" class="anchor"></a>Custom Filter</h4>
<p>Sometimes it is known a priori that certain peptides or samples should be filtered out of the dataset prior to statistical analysis. Perhaps there are known contaminant proteins, and so peptides mapping to them should be removed, or perhaps something went wrong during sample preparation for a particular sample. On the other hand, it may be preferred to specify peptides or samples to keep (removing those not explicitly specified), and this can also be accomplished. Keep in mind that both ‘remove’ and ‘keep’ arguments cannot be specified together; either ‘remove’ arguments only or ‘keep’ arguments only may be specified in a single call to custom_filter().</p>
<p>Here, we remove peptide with Mass Tag ID 1047 and sample <code>Infection 1</code> as an example.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">myfilter &lt;-<span class="st"> </span><span class="kw"><a href="../reference/custom_filter.html">custom_filter</a></span>(mypepData, <span class="dt">e_data_remove =</span> <span class="dv">1047</span>, <span class="dt">e_meta_remove =</span> <span class="ot">NULL</span>, <span class="dt">f_data_remove =</span> <span class="st">"Infection1"</span>)
<span class="kw">summary</span>(myfilter)</code></pre></div>
<pre><code>## 
## Summary of Custom Filter
## 
##                       Filtered Remaining Total
## SampleIDs (f_data)           1        11    12
## Mass_Tag_IDs (e_data)        1     15590 15591
## Proteins (e_meta)            0      2511  2511</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">mypepData_temp &lt;-<span class="st"> </span><span class="kw">applyFilt</span>(<span class="dt">filter_object =</span> myfilter, mypepData)
<span class="kw">summary</span>(mypepData_temp)</code></pre></div>
<pre><code>##                                     
## Class                        pepData
## Unique SampleIDs (f_data)         11
## Unique Mass_Tag_IDs (e_data)   15590
## Unique Proteins (e_meta)        2511
## Missing Observations           58431
## Proportion Missing             0.312
## Samples per group: Infection       8
## Samples per group: Mock            3</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">rm</span>(mypepData_temp)</code></pre></div>
<p>Here, we keep the samples <code>Infection 1</code>, <code>Infection 2</code>, <code>Infection 3</code>, <code>Infection 4</code> and <code>Infection 5</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">myfilter2&lt;-<span class="st"> </span><span class="kw"><a href="../reference/custom_filter.html">custom_filter</a></span>(mypepData, <span class="dt">e_data_keep =</span> <span class="ot">NULL</span>, <span class="dt">e_meta_keep =</span> <span class="ot">NULL</span>, <span class="dt">f_data_keep =</span> <span class="kw">c</span>(<span class="st">"Infection1"</span>, <span class="st">"Infection2"</span>, <span class="st">"Infection3"</span>, <span class="st">"Infection4"</span>, <span class="st">"Infection5"</span>))
<span class="kw">summary</span>(myfilter2)</code></pre></div>
<pre><code>## 
## Summary of Custom Filter
## 
##                        Kept Discarded Total
## SampleIDs (f_data)        5         7    12
## Mass_Tag_IDs (e_data) 15591         0 15591
## Proteins (e_meta)      2511         0  2511</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">mypepData_temp2&lt;-<span class="st"> </span><span class="kw">applyFilt</span>(<span class="dt">filter_object =</span> myfilter2, mypepData)
<span class="kw">summary</span>(mypepData_temp2)</code></pre></div>
<pre><code>##                                     
## Class                        pepData
## Unique SampleIDs (f_data)          5
## Unique Mass_Tag_IDs (e_data)   15591
## Unique Proteins (e_meta)        2511
## Missing Observations           25005
## Proportion Missing             0.267
## Samples per group: Infection       5</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">rm</span>(mypepData_temp2)</code></pre></div>
<p>Note that there is a <code>summary()</code> method for objects of type custom_filt, but no <code>plot()</code> method.</p>
</div>
<div id="imd-anova-filter" class="section level4">
<h4 class="hasAnchor">
<a href="#imd-anova-filter" class="anchor"></a>IMD-ANOVA Filter</h4>
<p>The IMD-ANOVA filter removes peptides that do not have sufficient data for the available statistical tests in the pmartRstat package. The statistical tests available are ANOVA (quantitative test) and independence of missing data (IMD) using a g-test (qualitative test). When using the <code>summary()</code>, <code>plot()</code>, and <code>applyFilt()</code> functions, you can specify just one filter (ANOVA or IMD) or both, depending on the tests you’d like to perform later. Using this filter speeds up the process of performing the statistical tests.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">myfilter &lt;-<span class="st"> </span><span class="kw"><a href="../reference/imdanova_filter.html">imdanova_filter</a></span>(mypepData)</code></pre></div>
<p>Here we consider what the filter would look like for both ANOVA and IMD.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">summary</span>(myfilter, <span class="dt">min_nonmiss_anova =</span> <span class="dv">2</span>, <span class="dt">min_nonmiss_gtest =</span> <span class="dv">3</span>)</code></pre></div>
<pre><code>## 
## Summary of IMD Filter
##                           
## Total Observations:  15591
## Filtered:             2052
## Not Filtered:        13539</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(myfilter, <span class="dt">min_nonmiss_anova =</span> <span class="dv">2</span>, <span class="dt">min_nonmiss_gtest =</span> <span class="dv">3</span>, <span class="dt">x_lab_size =</span> <span class="dv">9</span>, <span class="dt">y_lab_size =</span> <span class="dv">9</span>)</code></pre></div>
<pre><code>## Warning: Ignoring unknown aesthetics: fill</code></pre>
<p><img src="Quality_Control_with_pmartR_files/figure-html/imdanova_filter2-1.png" width="480"></p>
<p>Here we consider what the filter would look like for just ANOVA.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">summary</span>(myfilter, <span class="dt">min_nonmiss_anova =</span> <span class="dv">2</span>)</code></pre></div>
<pre><code>## 
## Summary of IMD Filter
##                           
## Total Observations:  15591
## Filtered:             5511
## Not Filtered:        10080</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(myfilter, <span class="dt">min_nonmiss_anova =</span> <span class="dv">2</span>)</code></pre></div>
<p><img src="Quality_Control_with_pmartR_files/figure-html/imdanova_filter3-1.png" width="480"></p>
<p>Here we consider what the filter would look like for just IMD.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">summary</span>(myfilter, <span class="dt">min_nonmiss_gtest =</span> <span class="dv">3</span>)</code></pre></div>
<pre><code>## 
## Summary of IMD Filter
##                           
## Total Observations:  15591
## Filtered:             2280
## Not Filtered:        13311</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(myfilter, <span class="dt">min_nonmiss_gtest =</span> <span class="dv">3</span>)</code></pre></div>
<p><img src="Quality_Control_with_pmartR_files/figure-html/imdanova_filter4-1.png" width="480"></p>
<p>Now we apply the filter for both ANOVA and IMD.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">mypepData &lt;-<span class="st"> </span><span class="kw">applyFilt</span>(<span class="dt">filter_object =</span> myfilter, mypepData, <span class="dt">min_nonmiss_anova =</span> <span class="dv">2</span>, <span class="dt">min_nonmiss_gtest =</span> <span class="dv">3</span>)

<span class="kw">summary</span>(mypepData)</code></pre></div>
<pre><code>##                                     
## Class                        pepData
## Unique SampleIDs (f_data)         12
## Unique Mass_Tag_IDs (e_data)   13539
## Unique Proteins (e_meta)        2253
## Missing Observations           42875
## Proportion Missing             0.244
## Samples per group: Infection       9
## Samples per group: Mock            3</code></pre>
</div>
</div>
<div id="filter-samples" class="section level3">
<h3 class="hasAnchor">
<a href="#filter-samples" class="anchor"></a>Filter Samples</h3>
<p>To identify any samples that are potential outliers or anomalies (due to sample quality, preparation, or processing circumstances), we use a robust Mahalanobis distance (rMd) <span class="citation">(Matzke et al. 2011)</span> score based on 2-5 metrics. The possible metrics are:</p>
<ul>
<li><p>Correlation</p></li>
<li><p>Proportion of data that is missing (“Proportion_Missing”)</p></li>
<li><p>Median absolute deviation (“MAD”)</p></li>
<li><p>Skewness</p></li>
<li><p>Kurtosis</p></li>
</ul>
<p>The rMd score can be mapped to a p-value, and a p-value threshold used to identify potentially outlying samples. In general, for proteomics data we recommend using correlation, proportion missing, MAD, and skew.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">myfilter &lt;-<span class="st"> </span><span class="kw">rmd_filter</span>(mypepData, <span class="dt">metrics =</span> <span class="kw">c</span>(<span class="st">"Correlation"</span>, <span class="st">"Proportion_Missing"</span>, <span class="st">"MAD"</span>, <span class="st">"Skewness"</span>))

<span class="kw">plot</span>(myfilter)</code></pre></div>
<p><img src="Quality_Control_with_pmartR_files/figure-html/rmd1-1.png" width="480"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(myfilter, <span class="dt">title_size=</span><span class="dv">7</span>, <span class="dt">bw_theme=</span><span class="ot">TRUE</span>)</code></pre></div>
<p><img src="Quality_Control_with_pmartR_files/figure-html/rmd1-2.png" width="480"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">summary</span>(myfilter, <span class="dt">pvalue_threshold =</span> <span class="fl">0.001</span>)</code></pre></div>
<pre><code>## 
## Summary of RMD Filter
## ----------------------
## P-values:
## 
##      Min.   1st Qu.    Median      Mean   3rd Qu.      Max. 
## 0.0000001 0.2686711 0.4152817 0.5050117 0.8306939 0.9707249 
## 
## Metrics Used: MAD, Skewness, Corr, Proportion_Missing 
## 
## Filtered Samples: Infection5, Infection8</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(myfilter, <span class="dt">pvalue_threshold =</span> <span class="fl">0.001</span>, <span class="dt">bw_theme=</span><span class="ot">TRUE</span>)</code></pre></div>
<p><img src="Quality_Control_with_pmartR_files/figure-html/rmd1-3.png" width="480"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(myfilter, <span class="dt">pvalue_threshold =</span> <span class="fl">0.001</span>, <span class="dt">bw_theme=</span><span class="ot">TRUE</span>, <span class="dt">legend_position=</span><span class="st">"bottom"</span>)</code></pre></div>
<p><img src="Quality_Control_with_pmartR_files/figure-html/rmd1-4.png" width="480"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(myfilter, <span class="dt">pvalue_threshold =</span> <span class="fl">0.001</span>, <span class="dt">legend_position=</span><span class="st">"top"</span>)</code></pre></div>
<p><img src="Quality_Control_with_pmartR_files/figure-html/rmd1-5.png" width="480"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(myfilter, <span class="dt">pvalue_threshold =</span> <span class="fl">0.001</span>, <span class="dt">bw_theme=</span><span class="ot">TRUE</span>, <span class="dt">legend_position=</span><span class="st">"left"</span>)</code></pre></div>
<p><img src="Quality_Control_with_pmartR_files/figure-html/rmd1-6.png" width="480"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(myfilter, <span class="dt">pvalue_threshold =</span> <span class="fl">0.001</span>, <span class="dt">x_lab_size =</span> <span class="dv">8</span>, <span class="dt">y_lab_size =</span> <span class="dv">8</span>)</code></pre></div>
<p><img src="Quality_Control_with_pmartR_files/figure-html/rmd1-7.png" width="480"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(myfilter, <span class="dt">pvalue_threshold =</span> <span class="fl">0.001</span>, <span class="dt">x_lab_size =</span> <span class="dv">8</span>, <span class="dt">y_lab_size =</span> <span class="dv">8</span>, <span class="dt">point_size =</span> <span class="dv">6</span>)</code></pre></div>
<p><img src="Quality_Control_with_pmartR_files/figure-html/rmd1-8.png" width="480"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(myfilter, <span class="dt">pvalue_threshold =</span> <span class="fl">0.001</span>, <span class="dt">x_lab_size =</span> <span class="dv">8</span>, <span class="dt">y_lab_size =</span> <span class="dv">8</span>, <span class="dt">point_size =</span> <span class="dv">7</span>)</code></pre></div>
<p><img src="Quality_Control_with_pmartR_files/figure-html/rmd1-9.png" width="480"></p>
<p>Using the output from the <code>summary()</code> function, we can explore the outliers identified to see what metrics are driving their outlier-ness. Box plots for each metric are plotted, with the specified sample marked with an ‘X’.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(myfilter, <span class="dt">sampleID =</span> <span class="st">"Infection5"</span>)</code></pre></div>
<p><img src="Quality_Control_with_pmartR_files/figure-html/rmd2-1.png" width="480"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(myfilter, <span class="dt">sampleID =</span> <span class="st">"Infection5"</span>, <span class="dt">bw_theme=</span><span class="ot">TRUE</span>)</code></pre></div>
<p><img src="Quality_Control_with_pmartR_files/figure-html/rmd2-2.png" width="480"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(myfilter, <span class="dt">sampleID =</span> <span class="st">"Infection8"</span>, <span class="dt">title_size =</span> <span class="dv">10</span>, <span class="dt">y_lab_size=</span><span class="dv">8</span>)</code></pre></div>
<p><img src="Quality_Control_with_pmartR_files/figure-html/rmd2-3.png" width="480"></p>
<p>We can see that Infection5 has the highest MAD, low skewness, and almost the lowest correlation, but the lowest proportion missing compared to the other samples. Infection 8 has MAD close to the median MAD value, the highest skew, the lowest correlation, and the highest proportion of missing data. We can use this information to determine whether to remove either or both of these samples. Sometimes it is also useful to look at additional data summaries to inform outlier removal, such as a principal components plot or correlation heat map (below).</p>
</div>
<div id="normalize-data" class="section level3">
<h3 class="hasAnchor">
<a href="#normalize-data" class="anchor"></a>Normalize Data</h3>
<p>The next step in a typical workflow is to normalize the data.</p>
<p>Normalization approaches consist of a subset method and a normalization function <span class="citation">(Åstrand 2003)</span>,<span class="citation">(Gautier et al. 2004)</span>. Available subset methods include:</p>
<ul>
<li><p>Percentage of peptides present (PPP): Subset the data to peptides present in at least <span class="math inline">\(p\)</span> percent of samples. This subset method is specified as “ppp”.</p></li>
<li><p>Rank invariant peptides (RIP): First subset peptides to those present in every sample (e.g. complete peptides). Next, subject each peptide to a Kruskal Wallis test on group, and those peptides <em>not</em> significant at a given p-value threshold are retained as invariant. This subset method is specified as “rip”.</p></li>
<li><p>PPP-RIP: Rank invariant peptides among peptides present in a given percentage of samples. This subset method is specified as “ppp_rip”.</p></li>
<li><p>Top “l” order statistics (LOS): The peptides with intensities in the top “l” order statistics are retained. This subset method is specified as “los”.</p></li>
</ul>
<p>Available normalization functions include:</p>
<ul>
<li><p>Median centering: The sample-wise median (median of all peptides in a given sample) is subtracted from each peptide in the corresponding sample. This normalization method is specified as “median”.</p></li>
<li><p>Mean centering: The sample-wise mean (mean of all peptides in a given sample) is subtracted from each peptide in the corresponding sample. This normalization method is specified as “mean”.</p></li>
<li><p>Z-score transformation: The sample-wise mean (mean of all peptides in a given sample) is subtracted from each peptide, and the result is divided by the sample-wise standard deviation (standard deviation of all peptides in a given sample) in the corresponding sample. This normalization method is specified as “zscore”.</p></li>
<li><p>Median absolute deviation (MAD) transformation: The sample-wise median (median of all peptides in a given sample) is subtracted from each peptide, and the result is divided by the sample-wise MAD (MAD of all peptides in a given sample) in the corresponding sample. This normalization method is specified as “mad”.</p></li>
</ul>
<p>There are two ways to go about normalizing data in the pmartR package. If you know the normalization approach you’d like to use, you can directly specify it in the <code>normalize_data()</code> function. Or, if you want to see how compatible different normalization approaches are with your dataset, you can use the <code>spans()</code> function [UNDER DEVELOPMENT].</p>
<p>We can go straight to the normalization, if we know what approach we want to use.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Normalize using all peptides and median centering #</span>
norm_object &lt;-<span class="st"> </span><span class="kw"><a href="../reference/normalize_global.html">normalize_global</a></span>(<span class="dt">omicsData =</span> mypepData, <span class="dt">subset_fn =</span> <span class="st">"all"</span>, <span class="dt">norm_fn =</span> <span class="st">"median"</span>, <span class="dt">apply_norm =</span> <span class="ot">FALSE</span>, <span class="dt">backtransform =</span> <span class="ot">TRUE</span>)
norm_data &lt;-<span class="st"> </span><span class="kw"><a href="../reference/normalize_global.html">normalize_global</a></span>(<span class="dt">omicsData =</span> mypepData, <span class="dt">subset_fn =</span> <span class="st">"all"</span>, <span class="dt">norm_fn =</span> <span class="st">"median"</span>, <span class="dt">apply_norm =</span> <span class="ot">TRUE</span>, <span class="dt">backtransform =</span> <span class="ot">TRUE</span>)
<span class="kw">plot</span>(norm_data, <span class="dt">title_plot=</span><span class="st">"Normalized Data: Median Centering Using all Peptides"</span>, <span class="dt">title_size=</span><span class="dv">12</span>)</code></pre></div>
<p><img src="Quality_Control_with_pmartR_files/figure-html/norm2-1.png" width="480"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Normalize using RIP 0.2 and median centering #</span>
norm_object &lt;-<span class="st"> </span><span class="kw"><a href="../reference/normalize_global.html">normalize_global</a></span>(<span class="dt">omicsData =</span> mypepData, <span class="dt">subset_fn =</span> <span class="st">"rip"</span>, <span class="dt">params=</span><span class="kw">list</span>(<span class="dt">rip=</span><span class="fl">0.2</span>), <span class="dt">norm_fn =</span> <span class="st">"median"</span>, <span class="dt">apply_norm =</span> <span class="ot">FALSE</span>, <span class="dt">backtransform =</span> <span class="ot">TRUE</span>)
norm_data &lt;-<span class="st"> </span><span class="kw"><a href="../reference/normalize_global.html">normalize_global</a></span>(<span class="dt">omicsData =</span> mypepData, <span class="dt">subset_fn =</span> <span class="st">"rip"</span>, <span class="dt">params=</span><span class="kw">list</span>(<span class="dt">rip=</span><span class="fl">0.2</span>), <span class="dt">norm_fn =</span> <span class="st">"median"</span>, <span class="dt">apply_norm =</span> <span class="ot">TRUE</span>, <span class="dt">backtransform =</span> <span class="ot">TRUE</span>)
<span class="kw">plot</span>(norm_data, <span class="dt">title_plot=</span><span class="st">"Normalized Data: Median Centering Using RIP 0.2 Peptides"</span>, <span class="dt">title_size=</span><span class="dv">12</span>)</code></pre></div>
<p><img src="Quality_Control_with_pmartR_files/figure-html/norm2-2.png" width="480"></p>
<p>Below we use SPANS to select a normalization approach and then normalize the data. [UNDER DEVELOPMENT]</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># data_norm &lt;- normalize_data(mypepData, subset_fn="ppp.5", norm_fun="median", backtransform=TRUE)</span></code></pre></div>
</div>
<div id="summarize-data" class="section level3">
<h3 class="hasAnchor">
<a href="#summarize-data" class="anchor"></a>Summarize Data</h3>
<p>The pmartR package contains various methods for data summarization and exploration that can be used as part of the quality control process: numeric summaries and associated plots (via the <code><a href="../reference/edata_summary.html">edata_summary()</a></code> function), sequential projection pursuit principal components analysis (sppPCA) for dimension reduction (via the<code><a href="../reference/dim_reduction.html">dim_reduction()</a></code> function), and a correlation heat map (via the <code><a href="../reference/cor_result.html">cor_result()</a></code> function)<span class="citation">(Stacklies et al. 2007)</span>. As for missing data summarization there are also functions to collect and plot missing data information. Missing data collection and plotting functions include, <code><a href="../reference/missingval_result.html">missingval_result()</a></code>, <code>plot_missingval()</code>, <code><a href="../reference/missingval_scatterplot.html">missingval_scatterplot()</a></code> and <code><a href="../reference/missingval_heatmap.html">missingval_heatmap()</a></code>.</p>
<div id="numeric-summaries" class="section level4">
<h4 class="hasAnchor">
<a href="#numeric-summaries" class="anchor"></a>Numeric Summaries</h4>
<p>We can generate numeric summaries of our data by either sample or molecule. The <code><a href="../reference/edata_summary.html">edata_summary()</a></code> function computes the mean, standard deviation, median, percent of observations for which a value was observed, the minimum value, and the maximum value.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/edata_summary.html">edata_summary</a></span>(<span class="dt">omicsData =</span> norm_data, <span class="dt">by =</span> <span class="st">"sample"</span>, <span class="dt">groupvar =</span> <span class="ot">NULL</span>)</code></pre></div>
<pre><code>## mean
##        sample             mean
## 1  Infection1 21.7971083294077
## 2  Infection2  21.731366523126
## 3  Infection3 21.8032232443703
## 4  Infection4 21.8774087539977
## 5  Infection5 21.4257258525994
## 6        ----             ----
## 8  Infection8 22.4709412800765
## 9  Infection9  22.179932218701
## 10      Mock1 22.5630135430607
## 11      Mock2 22.4045599582422
## 12      Mock3  22.350225943675
## 
## std_div
##        sample               sd
## 1  Infection1 2.41455648093841
## 2  Infection2 2.31224701980782
## 3  Infection3 2.35043331396441
## 4  Infection4 2.44617068574648
## 5  Infection5  2.5535287332081
## 6        ----             ----
## 8  Infection8 2.26180598552078
## 9  Infection9 2.36212527529442
## 10      Mock1 2.28423295890622
## 11      Mock2 2.18183571765243
## 12      Mock3 2.28027627057377
## 
## median
##        sample           median
## 1  Infection1 21.7706905193056
## 2  Infection2 21.6715407709563
## 3  Infection3 21.7490538629397
## 4  Infection4 21.8551805308581
## 5  Infection5  21.465407352971
## 6        ----             ----
## 8  Infection8 22.4154539852953
## 9  Infection9  22.122666472151
## 10      Mock1 22.5353174031891
## 11      Mock2 22.3226552138985
## 12      Mock3 22.3085808857998
## 
## pct_obs
##        sample           pct_obs
## 1  Infection1 0.796513775020312
## 2  Infection2 0.790014033532757
## 3  Infection3 0.730334588965212
## 4  Infection4 0.691705443533496
## 5  Infection5 0.802348770219366
## 6        ----              ----
## 8  Infection8 0.529359627742078
## 9  Infection9 0.611123421227565
## 10      Mock1 0.789940172834035
## 11      Mock2 0.796661496417756
## 12      Mock3 0.799098899475589
## 
## minimum
##        sample              min
## 1  Infection1 12.1066097831752
## 2  Infection2 12.3753923747856
## 3  Infection3 13.4885281669697
## 4  Infection4 12.2536652266729
## 5  Infection5  11.026508295961
## 6        ----             ----
## 8  Infection8 14.3702269404971
## 9  Infection9 13.4319078573245
## 10      Mock1 13.0850982141284
## 11      Mock2 13.6650934960137
## 12      Mock3 11.6186174442457
## 
## maximum
##        sample              max
## 1  Infection1 30.9532147548287
## 2  Infection2 30.7628477282651
## 3  Infection3 30.8984015884978
## 4  Infection4 31.2420886871443
## 5  Infection5 30.9490717160277
## 6        ----             ----
## 8  Infection8 32.3710788536963
## 9  Infection9 32.1531258831469
## 10      Mock1  31.275411454618
## 11      Mock2 31.2032188575782
## 12      Mock3 31.1871443967351</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#edata_summary(omicsData = norm_data, by = "sample", groupvar = "Condition")</span>

<span class="co">#edata_summary(omicsData = norm_data, by = "molecule", groupvar = NULL)</span></code></pre></div>
</div>
<div id="probabilistic-principal-components-analysis" class="section level4">
<h4 class="hasAnchor">
<a href="#probabilistic-principal-components-analysis" class="anchor"></a>Probabilistic Principal Components Analysis</h4>
<p>Probabilistic is a PCA algorithm that can be used in the presence of missing data. We use the <code><a href="../reference/dim_reduction.html">dim_reduction()</a></code> function in pmartR, specifying the dataset and the number of principal components to return (default value of 2 principal components). The <code>summary()</code> and <code>plot()</code> functions operate on the results of the <code><a href="../reference/dim_reduction.html">dim_reduction()</a></code> function.</p>
<p>Note that <code><a href="../reference/dim_reduction.html">dim_reduction()</a></code> requires that the <code><a href="../reference/group_designation.html">group_designation()</a></code> function has been run.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">pca_results &lt;-<span class="st"> </span><span class="kw"><a href="../reference/dim_reduction.html">dim_reduction</a></span>(norm_data, <span class="dt">k =</span> <span class="dv">2</span>)

<span class="kw">summary</span>(pca_results) </code></pre></div>
<pre><code>## Summary of 'dimRes' Object</code></pre>
<pre><code>##     R_squaredPC1     0.554PC2     0.329</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(pca_results)</code></pre></div>
<p><img src="Quality_Control_with_pmartR_files/figure-html/sppPCA-1.png" width="480"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(pca_results, <span class="dt">title_size =</span> <span class="dv">20</span>, <span class="dt">bw_theme=</span><span class="ot">TRUE</span>, <span class="dt">legend_position=</span><span class="st">"left"</span>)</code></pre></div>
<p><img src="Quality_Control_with_pmartR_files/figure-html/sppPCA-2.png" width="480"></p>
</div>
<div id="correlation-heatmap" class="section level4">
<h4 class="hasAnchor">
<a href="#correlation-heatmap" class="anchor"></a>Correlation Heatmap</h4>
<p>Pearson correlation between samples is calculated based on pairwise complete biomolecules.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">correlation_results &lt;-<span class="st"> </span><span class="kw"><a href="../reference/cor_result.html">cor_result</a></span>(norm_data)

<span class="kw">summary</span>(correlation_results)</code></pre></div>
<pre><code>## Summary of Correlation Matrix (Upper Triangle)
## 
##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##  0.7804  0.8688  0.9143  0.9032  0.9496  0.9722</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(correlation_results)</code></pre></div>
<p><img src="Quality_Control_with_pmartR_files/figure-html/corr-1.png" width="480"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(correlation_results, <span class="dt">title_size =</span> <span class="dv">20</span>)</code></pre></div>
<p><img src="Quality_Control_with_pmartR_files/figure-html/corr-2.png" width="480"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(correlation_results, <span class="dt">x_lab=</span><span class="ot">FALSE</span>, <span class="dt">title_plot=</span><span class="st">"My Correlation Heatmap"</span>)</code></pre></div>
<p><img src="Quality_Control_with_pmartR_files/figure-html/corr-3.png" width="480"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(correlation_results, <span class="dt">x_lab=</span><span class="ot">FALSE</span>, <span class="dt">y_lab=</span><span class="ot">FALSE</span>)</code></pre></div>
<p><img src="Quality_Control_with_pmartR_files/figure-html/corr-4.png" width="480"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(correlation_results, <span class="dt">colorbar_lim=</span><span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">1</span>))</code></pre></div>
<p><img src="Quality_Control_with_pmartR_files/figure-html/corr-5.png" width="480"></p>
</div>
<div id="missing-data" class="section level4">
<h4 class="hasAnchor">
<a href="#missing-data" class="anchor"></a>Missing Data</h4>
<p>Patterns of missing data can be explored using the <code><a href="../reference/missingval_result.html">missingval_result()</a></code> and <code>plot_missingval()</code> functions.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">results &lt;-<span class="st"> </span><span class="kw"><a href="../reference/missingval_result.html">missingval_result</a></span>(mypepData)


<span class="kw">plot</span>(results, <span class="dt">type =</span> <span class="st">"bySample"</span>)</code></pre></div>
<p><img src="Quality_Control_with_pmartR_files/figure-html/missingval-1.png" width="480"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(results, <span class="dt">type =</span> <span class="st">"bySample"</span>, <span class="dt">title_plot=</span><span class="st">"Hey look at this!"</span>)</code></pre></div>
<p><img src="Quality_Control_with_pmartR_files/figure-html/missingval-2.png" width="480"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(results, <span class="dt">type =</span> <span class="st">"byMolecule"</span>, <span class="dt">x_lab_angle =</span> <span class="dv">0</span>)</code></pre></div>
<p><img src="Quality_Control_with_pmartR_files/figure-html/missingval-3.png" width="480"></p>
<p>In addition, the <code><a href="../reference/missingval_scatterplot.html">missingval_scatterplot()</a></code> and <code><a href="../reference/missingval_heatmap.html">missingval_heatmap()</a></code> functions provide more views of the missing data.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/missingval_scatterplot.html">missingval_scatterplot</a></span>(mypepData, <span class="dt">palette =</span> <span class="st">"Set1"</span>)</code></pre></div>
<pre><code>## Warning: Removed 2204 rows containing missing values (geom_point).</code></pre>
<p><img src="Quality_Control_with_pmartR_files/figure-html/missingval_plots-1.png" width="480"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/missingval_scatterplot.html">missingval_scatterplot</a></span>(mypepData, <span class="dt">title_plot=</span><span class="st">"Hey look at this!"</span>, <span class="dt">palette =</span> <span class="st">"Set2"</span>)</code></pre></div>
<pre><code>## Warning: Removed 2204 rows containing missing values (geom_point).</code></pre>
<p><img src="Quality_Control_with_pmartR_files/figure-html/missingval_plots-2.png" width="480"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/missingval_heatmap.html">missingval_heatmap</a></span>(mypepData)</code></pre></div>
<p><img src="Quality_Control_with_pmartR_files/figure-html/missingval_plots-3.png" width="480"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/missingval_heatmap.html">missingval_heatmap</a></span>(mypepData, <span class="dt">title_size =</span> <span class="dv">20</span>, <span class="dt">title_plot =</span> <span class="st">"Hey look at this!"</span>)</code></pre></div>
<p><img src="Quality_Control_with_pmartR_files/figure-html/missingval_plots-4.png" width="480"></p>
</div>
</div>
<div id="protein-quantification-methods" class="section level3">
<h3 class="hasAnchor">
<a href="#protein-quantification-methods" class="anchor"></a>Protein Quantification Methods</h3>
<p>Protein quantification can be done using the <code><a href="../reference/protein_quant.html">protein_quant()</a></code> function, either with or without accounting for different isoforms of the proteins, also called ‘proteoforms’. Regardless of whether the user is accounting for protein isoforms, they must specify a method for rolling peptides up to proteins (one of “rollup”, “rollup”, “qrollup”, or “zrollup”) and a function to use for combining the peptide-level data (either “mean” or “median”). When the user does wish to account for protein isoforms, they have two options for doing so: Bayesian Proteoform Quantification (BP-Quant) <span class="citation">(Webb-Robertson et al. 2014)</span> or Protein Quantification via Peptide Quality-Control (PQPQ).</p>
<p>The <code><a href="../reference/protein_quant.html">protein_quant()</a></code> function returns an object of class proData.</p>
<div id="rollup-methods" class="section level4">
<h4 class="hasAnchor">
<a href="#rollup-methods" class="anchor"></a>Rollup Methods</h4>
<p>The rollup method …</p>
<p>In the rrollup method, all peptides that map to a single protein are scaled based on a chosen reference peptide, which is the peptide with most observations. Next the average or mean of the scaled peptides is set as the relative protein abundance. <span class="citation">(Matzke et al. 2013)</span></p>
<p>The qrollup method starts with all peptides that map to a single protein. Next peptides are chosen according to an abundance cutoff value and the average or mean of the scaled peptides is set as the protein abundance.</p>
<p>In the zrollup method, scaling is done similarly to the z-score formula (except with medians instead of means). The scaling formula is applied to peptides that map to a single protein and then the mean or median of the scaled peptides is set as protein abundance (from DAnTE article). The rollup method is similar to rrollup method, except there is no scaling involved in these methods. Either the mean or median is applied to all peptides that map to a single protein to obtain protein abundance. <span class="citation">(Polpitiya et al. 2008)</span></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">print</span>(mypepData)

mypepdata_median&lt;-<span class="st"> </span><span class="kw"><a href="../reference/protein_quant.html">protein_quant</a></span>(<span class="dt">pepData =</span> mypepData, <span class="dt">method =</span> <span class="st">"rollup"</span>, <span class="dt">combine_fn =</span> <span class="st">"median"</span>)
<span class="kw">print</span>(mypepdata_median)
<span class="kw">rm</span>(mypepdata_median)

mypepdata_rrollup&lt;-<span class="st"> </span><span class="kw"><a href="../reference/protein_quant.html">protein_quant</a></span>(<span class="dt">pepData =</span> mypepData, <span class="dt">method =</span> <span class="st">"rrollup"</span>)
<span class="kw">print</span>(mypepdata_rrollup)
<span class="kw">rm</span>(mypepdata_rrollup)</code></pre></div>
</div>
</div>
</div>
<div id="references" class="section level2 unnumbered">
<h2 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h2>
<div id="refs" class="references">
<div id="ref-ahmed1995pooling">
<p>Ahmed, SE. 1995. “A Pooling Methodology for Coefficient of Variation.” <em>Sankhyā: The Indian Journal of Statistics, Series B</em>. JSTOR, 57–75.</p>
</div>
<div id="ref-aastrand2003contrast">
<p>Åstrand, Magnus. 2003. “Contrast Normalization of Oligonucleotide Arrays.” <em>Journal of Computational Biology</em> 10 (1). Mary Ann Liebert, Inc.: 95–102.</p>
</div>
<div id="ref-gautier2004affy">
<p>Gautier, Laurent, Leslie Cope, Benjamin M Bolstad, and Rafael A Irizarry. 2004. “Affy—analysis of Affymetrix Genechip Data at the Probe Level.” <em>Bioinformatics</em> 20 (3). Oxford University Press: 307–15.</p>
</div>
<div id="ref-matzke2013comparative">
<p>Matzke, Melissa M, Joseph N Brown, Marina A Gritsenko, Thomas O Metz, Joel G Pounds, Karin D Rodland, Anil K Shukla, et al. 2013. “A Comparative Analysis of Computational Approaches to Relative Protein Quantification Using Peptide Peak Intensities in Label-Free Lc-Ms Proteomics Experiments.” <em>Proteomics</em> 13 (3-4). Wiley Online Library: 493–503.</p>
</div>
<div id="ref-matzke2011improved">
<p>Matzke, Melissa M, Katrina M Waters, Thomas O Metz, Jon M Jacobs, Amy C Sims, Ralph S Baric, Joel G Pounds, and Bobbie-Jo M Webb-Robertson. 2011. “Improved Quality Control Processing of Peptide-Centric Lc-Ms Proteomics Data.” <em>Bioinformatics</em> 27 (20). Oxford University Press: 2866–72.</p>
</div>
<div id="ref-polpitiya2008dante">
<p>Polpitiya, Ashoka D, Wei-Jun Qian, Navdeep Jaitly, Vladislav A Petyuk, Joshua N Adkins, David G Camp, Gordon A Anderson, and Richard D Smith. 2008. “DAnTE: A Statistical Tool for Quantitative Analysis of-Omics Data.” <em>Bioinformatics</em> 24 (13). Oxford University Press: 1556–8.</p>
</div>
<div id="ref-stacklies2007pcamethods">
<p>Stacklies, Wolfram, Henning Redestig, Matthias Scholz, Dirk Walther, and Joachim Selbig. 2007. “PcaMethods—a Bioconductor Package Providing Pca Methods for Incomplete Data.” <em>Bioinformatics</em> 23 (9). Oxford University Press: 1164–7.</p>
</div>
<div id="ref-webb2014bayesian">
<p>Webb-Robertson, Bobbie-Jo M, Melissa M Matzke, Susmita Datta, Samuel H Payne, Jiyun Kang, Lisa M Bramer, Carrie D Nicora, et al. 2014. “Bayesian Proteoform Modeling Improves Protein Quantification of Global Proteomic Measurements.” <em>Molecular &amp; Cellular Proteomics</em> 13 (12). ASBMB: 3639–46.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#data">Data</a></li>
      <li><a href="#workflow">Workflow</a></li>
      <li><a href="#references">References</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by .</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
